src/rabbitalerter.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * An alerting system for message queues.
+ * @module rabbitalerter
+ */
+const assert = require('assert');
+const taskcluster = require('taskcluster-client');
+
+/**
+ * An interface used to generate various TaskCluster Notifications.
+ *
+ * @class RabbitAlerter
+ */
+class RabbitAlerter {
+  /**
+   * @param {number} config.alerter.messageCountTolerance       - When a queue holds more messages
+   *                                                              than this tolerance, an alert will be created.
+   * @param {number} config.alerter.messagePublishRateTolerance - When the number of messages published per second
```

Comment:
> when the publishing rate of messages to a queue is higher than that queue's consumption rate

Yeah, but that is easier said than done... Because "consumption rate" will never be higher than "publishing rate". And "publishing rate" should definitely be allowed to spike :)

> We also do not want to send too many false positive alerts. For example, a queue has zero consumption rate, and one message is being published to it a day.

Yeah, probably we want some threshold to say if queue has less than 100 messages we never alert... Unless it's been like that for a day..

> We can see how many messages were consumed by keeping track of the previous queue snapshot (each snapshot will show how many messages the queue has and when). We know that messages were consumed when the current snapshot shows less messages than the previous one.

Well, messages are both published and consumed in-between snapshots, so total number of messages in the queue is of little use...

If you look at: https://gist.github.com/jonasfj/925b8652b015ff857c629cef6106fac7
Then perhaps `message_stats` can be used to figure out number of messages published and consumed between two snapshots...

Last @imbstack and I talked about it, our thinking was that what we really wanted was to know how old the oldest message in the queue was and alert if it was older than say 15 minutes. Maybe even delete the queue if older than 7 days and nobody reacted on it...

So just brainstorming here... but what if we stored the `message_stats.publish` and `message_stats.ack` (acknowledged messages are those fully processed) for each minute for the past say 30 minutes or so?
Wouldn't we then get something like:
```
time: 13:00  published:   0  acked:    0  queue:  0  oldest: 13.00  max-age: 0 min
time: 13:01  published:  10  acked:    5  queue:  5  oldest: 13.01  max-age: 0 min
time: 13:02  published:  50  acked:   20  queue: 30  oldest: 13.02  max-age: 0 min
time: 13:03  published:  50  acked:   30  queue: 20  oldest: 13.02  max-age: 1 min
time: 13:04  published:  50  acked:   35  queue: 15  oldest: 13.02  max-age: 2 min
time: 13:05  published:  50  acked:   40  queue: 10  oldest: 13.02  max-age: 3 min
time: 13:06  published:  50  acked:   45  queue:  5  oldest: 13.02  max-age: 4 min
time: 13:07  published:  60  acked:   50  queue: 10  oldest: 13.07  max-age: 0 min
time: 13:08  published:  60  acked:   60  queue:  0  oldest: 13.08  max-age: 0 min
```

Wouldn't we then be able to say at `13:06` the oldest message in the queue was ~4 minutes old.
Similarly, couldn't we store the published and acked numbers for a snapshot every day... Enabling us to see if all messages published yesterday have been consumed.

Let's say we keep:
 A) a `{published: ..., acked: ...}` snapshot for each minute for the last 30 minutes.
 B) a `{published: ..., acked: ...}` for the start of each day for the past 7 days.

Then we can alarm if:
  * All messages older than 30 min haven't been consumed (using (A))  
  * All messages older than 3 days haven't been consumed (using (B))
  * All messages older than 4 days haven't been consumed (using (B))
  * All messages older than 5 days haven't been consumed (using (B))
  * All messages older than 6 days haven't been consumed (using (B))

And finally, we can delete the queue if:
 1. all messages older than 7 days haven't been consumed, and
 2. the queue contains more than 1000 messages.

I suggest adding condition (2) because there is no real value in deleting queues with less than 1000 messages. They aren't going to cause us problems.

Maybe we need an additional delete scenario like: delete queue if:
 1. all messages older than 2 days haven't been consumed, and
 2. the queue contains more than 100k messages.
Based on the assumption that 100k messages might start to eat up system resources.